packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -893,16 +907,27 @@ class IOSDeviceLogReader extends DeviceLogReader {
   // Strip off the logging metadata (leave the category), or just echo the line.
   String _debuggerLineHandler(String line) => _debuggerLoggingRegex.firstMatch(line)?.group(1) ?? line;
 
+  /// Use both logs from `idevicesyslog` and `ios-deploy` when debugging from a CI bot
+  /// since sometimes `ios-deploy` does not return the Dart VM url:
+  /// https://github.com/flutter/flutter/issues/121231
+  bool get useBothLogDeviceReaders {
+    final String? user = _platform.environment['USER'];
+    if (user != null && user == 'swarming' && _majorSdkVersion >= 16) {
```

Comment:
I tried to target just our CI bots by checking if the user is `swarming` but I'm unsure if this is a good way to do it.

Reply:
can you instead add a helper method to `class FlutterCommand`, that looks something like this:

```dart
void usesCiFlag() {
  argParser.addFlag(
    'ci',
    defaultsTo: false,
    help: 'Enable a set of CI-specific test debug settings.',
    hide: !verboseHelp,
  )
}
```

And then call that from the relevant sub-commands. You'll need to also update whatever tests/runners invoke the flutter CLI tool, but this should let us keep these types of settings organized moving forward.

Reply:
Somewhere around here: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L273

Reply:
What do you think about it being a global option here? 
https://github.com/flutter/flutter/blob/b4796915b3dc0dfac117f7c68ea057c972088ff2/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L24-L128

Our testing suite tests basically every flutter command, so if we want to make it more usable in the future for other purposes, I think that might be the way to go. However, it then begs the question of how to actually get the value. That could be per sub-command to do for itself or it could be a global/context thing. What do you think? I believe we're trying to move away from globals?